docs/dev/windows.md

```diff
@@ -622,133 +690,138 @@ input
 </div>
 </div>
 
-## Dealing with Late Data
+## Triggers
 
-When working with event-time windowing it can happen that elements arrive late, i.e the
-watermark that Flink uses to keep track of the progress of event-time is already past the
-end timestamp of a window to which an element belongs. Please
-see [event time](./event_time.html) and especially
-[late elements](./event_time.html#late-elements) for a more thorough discussion of
-how Flink deals with event time.
+A `Trigger` determines when a window (as formed by the `WindowAssigner`) is ready to be
+processed by the *window function*. Each `WindowAssigner` comes with a default `Trigger`. 
+If the default trigger does not fit your needs, you can specify a custom trigger using `trigger(...)`.
 
-You can specify how a windowed transformation should deal with late elements and how much lateness
-is allowed. The parameter for this is called *allowed lateness*. This specifies by how much time
-elements can be late. Elements that arrive within the allowed lateness are still put into windows
-and are considered when computing window results. If elements arrive after the allowed lateness they
-will be dropped. Flink will also make sure that any state held by the windowing operation is garbage
-collected once the watermark passes the end of a window plus the allowed lateness.
+The trigger interface provides five methods that react to different events: 
```

Comment:
I think it doesn't "provide" the methods, maybe write "The trigger interface has five methods that allow a Trigger to react to different events"